File: VB/Contacts/Reports/PersonReportService.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.XtraReports.UI

Namespace Contacts.Reports
	Public Class PersonReportService
        Implements ReportTypeService
		Private Shared _counter As Integer = 0
        Public Function GetDefaultReport() As XtraReport Implements ReportTypeService.GetDefaultReport
            _counter += 1
            Dim r As New PersonReport()
            r.Name = String.Format("PersonReport{0}", _counter)
            r.DisplayName = r.Name
            r.LoadDataSource()
            Return r
        End Function

        Public Function [GetType](ByVal reportType As Type) As Type Implements ReportTypeService.GetType
            Return reportType
        End Function
	End Class
End Namespace

File: CS/Contacts/Reports/PersonReportService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.XtraReports.UI;

namespace Contacts.Reports
{
    public class PersonReportService : ReportTypeService
    {
        private static int _counter = 0;
        public XtraReport GetDefaultReport()
        {
            _counter++;
            PersonReport r = new PersonReport();
            r.Name = string.Format("PersonReport{0}", _counter);
            r.DisplayName = r.Name;
            r.LoadDataSource();
            return r;
        }

        public Type GetType(Type reportType)
        {
            return reportType;
        }
    }
}
